src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+        }
+
+        if (isLeapDay == isYearDay) {
+            throw new DateTimeException("Invalid date: " + prolepticYear + " both leap-day and year-day are " + isYearDay);
+        }
+
+        if (isLeapDay && !InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear)) {
+            throw new DateTimeException("Invalid leap day for year: " + prolepticYear);
+        }
+
+        return new InternationalFixedDate(prolepticYear, isLeapDay, isYearDay);
+    }
+
+    /**
+     *
+     * Validates the object.
+     *
+     * @return the resolved date, not null
+     */
+    private Object readResolve() {
+        if (isLeapDay() || isYearDay()) {
+            return InternationalFixedDate.of(prolepticYear, isLeapDay(), isYearDay());
+        }
+
+        return InternationalFixedDate.of(prolepticYear, getMonth(), getDayOfMonth());
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getProlepticYear() {
+        return prolepticYear;
+    }
+
+    public boolean isLeapDay() {
+        return isLeapYear() && dayOfYear == 169;
+    }
+
+    public boolean isYearDay() {
+        return dayOfYear == InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        int doy = getDayOfYearAdjusted() - 1;
+
+        return 1 + (doy / InternationalFixedChronology.DAYS_IN_MONTH);
+    }
+
+    /**
+     * For calculation purposes, consider leap day to be associated with month Sol.
+     * In the same spirit, associate year day with the last month of the year.
+     * @return
+     */
+    private int getCalculatedMonth() {
+        return isYearDay() ?
+            InternationalFixedChronology.MONTHS_IN_YEAR : isLeapDay() ? 7 : getMonth();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getDayOfMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        int doy = getDayOfYearAdjusted() - 1;
+
+        return 1 + (doy % InternationalFixedChronology.DAYS_IN_MONTH);
+    }
+
+    private int getCalculatedDayOfMonth() {
+        return isYearDay() ? InternationalFixedChronology.DAYS_IN_MONTH + 1 : isLeapDay() ? 1 : getDayOfMonth();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getDayOfYear() {
+        return dayOfYear;
+    }
+
+    private int getDayOfYearAdjusted() {
+        return isLeapYear() && dayOfYear > 6 * InternationalFixedChronology.DAYS_IN_MONTH ? dayOfYear - 1 : dayOfYear;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int lengthOfYearInMonths() {
+        return InternationalFixedChronology.MONTHS_IN_YEAR;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    ValueRange rangeAlignedWeekOfMonth() {
```

Comment:
Since leap-day/year-day aren't part of any actual month, you need to return the zero range